// @prisma/migrate/dist/utils/errors.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var errors_exports = {};
__export(errors_exports, {
  DbDropNeedsForceError: () => import_chunk_U374EXYQ.DbDropNeedsForceError,
  DbPushIgnoreWarningsWithFlagError: () => import_chunk_U374EXYQ.DbPushIgnoreWarningsWithFlagError,
  MigrateDevEnvNonInteractiveError: () => import_chunk_U374EXYQ.MigrateDevEnvNonInteractiveError,
  MigrateNeedsForceError: () => import_chunk_U374EXYQ.MigrateNeedsForceError,
  MigrateResetEnvNonInteractiveError: () => import_chunk_U374EXYQ.MigrateResetEnvNonInteractiveError,
  NoSchemaFoundError: () => import_chunk_U374EXYQ.NoSchemaFoundError
});
module.exports = __toCommonJS(errors_exports);
var import_chunk_U374EXYQ = require("../chunk-U374EXYQ.js");
var import_chunk_6TE2RIPN = require("../chunk-6TE2RIPN.js");
var import_chunk_WWAWV7DQ = require("../chunk-WWAWV7DQ.js");
(0, import_chunk_U374EXYQ.init_errors)();
